chainer/iterators/random_state.py

```diff
@@ -0,0 +1,89 @@
+import contextlib
+import threading
+
+import numpy
+
+
+_random_state = threading.local()
+_random_state.value = None
+
+
+def get_random_state():
+    """Get a :class:`~numpy.random.RandomState` during iteration.
```

Comment:
I think "during iteration" is not needed as users can call this function any time, although the most expected use case is to call it during iterations.

Reply:
It would be better to note that returned instance is a thread-local variable.